Replicator/Replicator.cc

```diff
@@ -182,12 +138,15 @@ namespace litecore { namespace repl {
             C4RemoteID remoteDBID = _db->lookUpRemoteDBID(key);
             logVerbose("Remote-DB ID %u found for target <%.*s>", remoteDBID, SPLAT(key));
 
-            for (CollectionIndex i = 0; i < _subRepls.size(); ++i) {
-                if (_options->push(i) > kC4Passive || _options->pull(i) > kC4Passive) {
-                    // Get the checkpoints:
-                    if (getLocalCheckpoint(reset, i)) {
-                        getRemoteCheckpoint(false, i);
-                    }
+            bool goOn = true;
+            for (CollectionIndex i = 0; goOn && i < _subRepls.size(); ++i) {
+                goOn = goOn && getLocalCheckpoint(reset, i);
+            }
+            if (goOn) {
```

Comment:
If not goOn, should be an error and stop as well? Not sure what will be the part after not goOn.

Reply:
c.f. getLocalCheckpoint() -- it returns false after "getError(), stop()" We cannot go on if any one of getLocalCheckpoint() fails.